test/e2e/gateways/create_test.go

```diff
@@ -0,0 +1,118 @@
+package gateways
+
+import (
+	"context"
+	"errors"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/spf13/cobra"
+
+	root "github.com/equinix/metal-cli/internal/cli"
+	"github.com/equinix/metal-cli/internal/gateway"
+	outputPkg "github.com/equinix/metal-cli/internal/outputs"
+	"github.com/equinix/metal-cli/test/helper"
+)
+
+func TestGateways_Create(t *testing.T) {
+	var projectId, deviceId string
+	subCommand := "gateways"
+	consumerToken := ""
+	apiURL := ""
+	Version := "devel"
+	rootClient := root.NewClient(consumerToken, apiURL, Version)
+
+	device, err := helper.SetupProjectAndDevice(t, &projectId, &deviceId)
+	defer func() {
```

Comment:
use `t.Cleanup()` refer [here](https://ieftimov.com/posts/testing-in-go-clean-tests-using-t-cleanup/#:~:text=Cleanup%20registers%20a%20function%20to,all%20the%20subtests%20are%20done.&text=of%20our%20tests%3A-,func%20createUser(t%20*testing.,T%2C%20db%20*gorm.)